DocsScraper/src/DocsScraper.jl

```diff
@@ -0,0 +1,152 @@
+"""
+    get_base_url(url::String)
+
+Extracts the base url.
+
+# Arguments
+- `url`: The url string of which, the base url needs to be extracted
+"""
+function get_base_url(url::String)
+    parsed_url = URIs.URI(url)
+    base_url = string(parsed_url.scheme, "://", parsed_url.host, parsed_url.port != nothing ? ":" * string(parsed_url.port) : "", parsed_url.path)
+    return base_url
+end
+
+
+"""
+    process_nodes(node, heading_hierarchy::Dict, parsed_text::Vector)
+
+Process the nodes recursively. The function recursively calls itself for every HTML hierarchy, thereby going deeper to retrieve the text.
+
+# Arguments
+- `node`: The root HTML node 
+- `heading_hierarchy`: Dictionary used to store metadata
+- `parsed_text`: Vector of Dicts to store parsed text and metadata
+
+# TODO:
+- Inline code blocks are the ones present inside <p> or <li>. put `` around the inline code blocks. Make changes inside `elseif tag_name == "p" || tag_name == "li"`
+"""
+function process_nodes(node, heading_hierarchy::Dict, parsed_text::Vector)
+    # if a node is an html element
+    if typeof(node) <: Gumbo.HTMLElement
```

Comment:
Good idea. If I do this then, `function process_nodes(node::Gumbo.HTMLText, ...) ... end ` doesn't require other parameters: `heading_hierarchy`, `parsed_text`. so to maintain clarity, I can do any one from the following:
1) Leave the unused `heading_hierarchy` and `parsed_text`
2) remove `heading_hierarchy` and `parsed_text`. If I do this, then I have to add an if condition while function calls
3) use args... with proper comments to represent the data structures passed.

Which method do you think would be the best?

Reply:
If you want to keep the same interface, 1 and 3 are possible. 

1) is slightly more strict so potentially more robust to accidental bugs in the future (and can be easily changed if needed), but 3) is correct as well.

I have done both. I think I tend to default to 3) because of convenience

Reply:
Gotch. Thanks!